content/en/tutorial/10-signals.md

```diff
@@ -0,0 +1,306 @@
+}
+```
+
+## Global state with Signals
+
+Global state is state which be accessed by **Multiple components**, to make a signal globally accesable,
+create it outside using `signal(initialValue)`, which makes the state reusable anywhere,
+and `computed(fn)`, which lets you define derived values which will update automatically when the signal changes.
+The simplest way to use global state is to directly import and use the signal across multiple components.
+
+For example, `const theme = signal("light")` could be toggled in one component and read in another.
+But this can become harder to manage in larger apps, there's a common way where the global state will wrap in a **Context**.
+With context, you can create a function like `createAppState()` that stores your signals and computed values.
+Then, you give this state to your whole app by wrapping it in `Context.Provider`.
+By This way, any component inside the app can easily access the shared state without passing it through props.
+Any component can then access this shared state using `useContext`.
+
+In short, using signal and computed directly as global state works well for simple apps,
+but using context is helpful when many components in different parts of your app need to use the same global state.
+
+```jsx
+import { signal, computed } from '@preact/signals';
+
+export const counter = signal(0);
+export const doubled = computed(() => counter.value * 2);
+
+function Increment() {
+	return <button onClick={() => counter.value++}>Increment</button>;
+}
+
+function Display() {
+	return <p>Counter doubled: {doubled}</p>;
+}
+```
+
+Here, `counter` and `doubled` are global signals that both components (Increment and Display) can use directly.
+
+```jsx
+import { signal, computed } from '@preact/signals';
+import { createContext } from 'preact';
+import { useContext } from 'preact/hooks';
+
+// Create a state object
+function createAppState() {
+	const counter = signal(0);
+	const doubled = computed(() => counter.value * 2);
+	return { counter, doubled };
+}
+
+// Create a context
+const AppState = createContext();
+
+// Component 1: Increment button
+function Increment() {
+	const state = useContext(AppState); // access shared state
+	return <button onClick={() => state.counter.value++}>Increment</button>;
+}
+
+// Component 2: Display result
+function Display() {
+	const state = useContext(AppState); // access shared state
+	return <p>Counter doubled: {state.doubled}</p>;
+}
+
+// Main App
+export function App() {
+	return (
+		<AppState.Provider value={createAppState()}>
+			<Increment />
+			<Display />
+		</AppState.Provider>
+	);
+}
+```
+
+Here, `counter` and `doubled` is created once inside `createAppState()` and passed to all components through `Context`,
+so you don’t need to import global signals directly.
+
+## Try it!
+
+Let’s create a counter which gets incremented while the button is clicked using `signal`, which is similar to the state example from fourth chapter.
```

Comment:
We can drop the mention of the fourth chapter, but I'd ask that we add a mention that the user should display the current count (`{count}`) -- I realize chapter 4 doesn't really mention that either, but that's a mistake & sort of confusing, IMO.